torch/csrc/tensor/python_tensor.cpp

```diff
@@ -277,6 +277,11 @@ static THPObjectPtr get_tensor_dict() {
 static std::vector<PyTensorType*> tensor_types;
 
 void set_default_tensor_type(PyTensorType* type) {
+  // Make sure this is not a null pointer before accessing it.
+  if (!type) {
```

Comment:
seems like this should be an internal assert (`TORCH_INTERNAL_ASSERT(type)`)

Reply:
Isn't `type` provided by the user here?

Reply:
I think this function is a helper function and `py_set_default_tensor_type` is what is called when users call https://pytorch.org/docs/master/generated/torch.set_default_tensor_type.html#torch.set_default_tensor_type https://github.com/pytorch/pytorch/blob/5e6f29661254e5ebc97fac3d829e5d455cda5864/torch/csrc/Module.cpp#L206-L212.

Reply:
Aha, that makes sense

Reply:
i mean, i think it's better to write code like this regardless of whether it's user-provided or not. because it might become user-provided later.

as for the assert, is that preferred over throw somehow?

Reply:
talked about this with mario, i'll switch this to the assert